### ISSUE TITLE
Add Operation

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1366]
#### Description:
As an industry user, given that I have access, I can add operations to my operator.
- [Design](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?node-id=8912-244&t=pmKjK1RmQeLYOb04-4)
- [Prototype](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?page-id=7692%3A2&type=design&node-id=7763-4636&viewport=153%2C235%2C0.04&t=9CHcuF7tZEcxkUP7-1&scaling=scale-down&starting-point-node-id=7692%3A3728)

#### Acceptance Criteria:
Given that the user has navigated to the add and register operation button on the Operations grid OR the register operation tile on Registration dashboard > Register an Operation > Operation information > Don’t see the operation? Add a new operation, then the Add Operation form appears on the Operation Information page
https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=7763-4664&t=rBpJCzhxRzIMrjwm-4

When I fill out the form passing the form validation and click the "Submit" button
Then I see the form saved
And if I go back to the Operations grid,
Then I see the newly added operation in the Operations grid. If I have not completed the rest of the steps in Register an Operation, then I see the registration status "Continue Registration" 

Given I click the Operation Information button on the Operations Grid 
I can see the information I added from Add Operation in View mode
https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=8471-1866&t=rBpJCzhxRzIMrjwm-4

#### Development Checklist:

Changes to make to `Operation` data model:
- [ ]  `naics_code` field: no changes required to field, but add comment that the column refers to "primary naics code"
- [ ] `secondary_naics_code` - add nullable field that is an FK to our naics codes table
- [ ] `tertiary_naics_code` - same as secondary_naics_code
- [ ] add many-to-many relationship between `Operation` and `Registration Category` (see below)

Changes to make to `document_type` data model:
- [ ] add records for `equipment_list` and `new_entrant_application`

Changes to make to `bc_obps_regulated_operation` data model:
- [ ] add `status` field (required) which is an enum of either "Active" or "Inactive". All statuses should be "Active" by default

Create new enum class `Registration Category`:
- [ ] options: "Potential Reporting Operation", "Reporting Operation", "Regulated Operation", "Opt-in Operation", "New Entrant Operation", "Electricity Import Operation"
(additional data fields specific to certain registration categories will be separated out into other tickets)

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by patrickisaac** [comment]
Zoey to split into two different tickets:

```
This is the part that can come out: Given that I am on operations page,
When I click view details on a row
Then I am brought to that operation's operation details page
[Figma link](https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=7781-6342&mode=design&t=KFDAnKMzwJ3b1l14-4)
```

**Comment by hannavovk** [comment]
@zoeyli-46 and/or @nanyangpro, can we also split out the grid from the forms into two separate tickets. I think it would make sense to make a new ticket for the operations grid and cut this back to just the form (i.e. the first AC would be its own ticket).

Also, in the 'new' operations grid ticket, we may want to either be clear that it covers just the operations part of the page and the facilities grid is a separate ticket, or alternatively, have this one new ticket covering the combined operations and facilities page (showing both grids)

**Comment by zoeyli-46** [comment]
Done, Operation grid is ticket [#1686](https://github.com/bcgov/cas-registration/issues/1686).  I split up the facilities grid into another ticket, because its a separate action, it is in #1360

**Comment by nanyangpro** [comment]
Design/Prototype links are added or updated with AC refined further.

**Comment by andrea-williams** [comment]
The Operation Representatives that are to be assigned to an operation are currently blocked/conflicting with ongoing work #1351 , so I recommend omitting the operation rep stuff from this ticket unless it's unblocked soon

**Comment by marcellmueller** [comment]
Closing as the `Add Operation` form flow is being replaced with the `Register an Operation` form. The work that had been done on this has been used for the `Operator Information` form since they are essentially the same.
